src/api/myanimelist.ts

```diff
@@ -0,0 +1,45 @@
+import * as request from "request";
+import rp from "request-promise";
+import { StringNullableChain } from "lodash";
+
+export enum MediumType {
+  anime = "anime",
+  manga = "manga"
+}
+
+export interface AnimeAndManga {
+  results: [
+    {
+      mal_id: number;
+      url: string;
+      image_url: string;
+      title: string;
+      airing: boolean;
+      publishing: boolean;
+      synopsis: string;
+      type: string;
+      episodes: number;
+      volumes: number;
+      chapters: number;
+      members: number;
+      score: number;
+      start_date: string;
+      end_date: string;
+    }
+  ];
+}
+
+export function getAnimeManga(type: MediumType, title: string): Promise<AnimeAndManga> {
+  return new Promise((resolve, reject) => {
+    const options = {
```

Comment:
Why are you creating the promise directly instead of using request-promise? If it's because of the JSON parsing, then you know you can add `json: true` to the options, and it will parse it as json for you?

So this could be:

```
export function getAnimeManga(type: MediumType, title: string): Promise<AnimeAndManga> {
    const options = {
      url: `https://api.jikan.moe/v3/search/${type}?q=${title}&page=1&limit=1`,
      json: true
     };
     return rp(options);
}
```